.travis.yml

```diff
@@ -50,25 +64,32 @@ matrix:
       env:
         - COMPILER=clang++ V='Apple LLVM 11.0'
         - COMPILER=clang++ V='Apple LLVM 11.0' WITH_CPP14=true
+    - os: osx
+      osx_image: xcode11.2
+      env:
+        - COMPILER=clang++ V='Apple LLVM 11.0'
+        - COMPILER=clang++ V='Apple LLVM 11.0' WITH_CPP17=true
 
 before_install:
-  # - |
-  #   if [[ "${TRAVIS_OS_NAME}" == "osx" ]]; then
-  #     brew update
-  #     brew rm --force cmake && brew install cmake
-  #   fi
-
   - CMAKE_CXX_FLAGS+=" -Wall"
-
-  - if [[ "${WITH_CPP14}" == "true" ]]; then CMAKE_OPTIONS+=" -DCMAKE_CXX_STANDARD=14"; fi
+  - |
+    if [[ "${WITH_CPP14}" == "true" ]]; then
+        CMAKE_OPTIONS+=" -DCMAKE_CXX_STANDARD=14"
+    fi
+  - |
+    if [[ "${WITH_CPP17}" == "true" ]]; then
+        CMAKE_OPTIONS+=" -DCMAKE_CXX_STANDARD=17"
+    fi
   - |
     if [[ "${USE_BOOST_REGEX}" == "ON" ]]; then
         CMAKE_OPTIONS+=" -DUSE_BOOST_REGEX=ON"
-        CMAKE_OPTIONS+=" -DBoost_REGEX_LIBRARY_DEBUG=/usr/lib/x86_64-linux-gnu/libboost_regex.so.1.55.0"
-        CMAKE_OPTIONS+=" -DBoost_REGEX_LIBRARY_RELEASE=/usr/lib/x86_64-linux-gnu/libboost_regex.so.1.55.0"
+        CMAKE_OPTIONS+=" -DBoost_REGEX_LIBRARY_DEBUG=/usr/lib/x86_64-linux-gnu/libboost_regex.so"
+        CMAKE_OPTIONS+=" -DBoost_REGEX_LIBRARY_RELEASE=/usr/lib/x86_64-linux-gnu/libboost_regex.so"
```

Comment:
It could be cleaner to do the linking inside the `CMakeLists.txt`. This could would also be simpler to port this file to appveyor

Reply:
Though it is a secondary issue that could be solved in a separate PR.

Reply:
These are defines that tell CMake where it can look for the installed boost libraries; this is specific to Travis so this would not belong to the CMake file (although I am curious why this is needed at all; open to additional futzing around on this in new PR).